internal/jujuapi/httpproxy_integration_test.go

```diff
@@ -0,0 +1,207 @@
+	"github.com/juju/juju/api"
+	"github.com/juju/juju/api/client/charms"
+	"github.com/juju/juju/api/client/resources"
+	"github.com/juju/juju/testcharms"
+	"github.com/juju/juju/testing/factory"
+	"github.com/juju/names/v5"
+	"github.com/juju/version/v2"
+	gc "gopkg.in/check.v1"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/jimmtest"
+	"github.com/canonical/jimm/v3/internal/jujuapi"
+	ofganames "github.com/canonical/jimm/v3/internal/openfga/names"
+)
+
+type httpProxySuite struct {
+	jimmtest.JIMMSuite
+	model *dbmodel.Model
+}
+
+var _ = gc.Suite(&httpProxySuite{})
+
+const testEnv = `
+clouds:
+- name: test-cloud
+  type: test-provider
+  regions:
+  - name: test-cloud-region
+cloud-credentials:
+- owner: alice@canonical.com
+  name: cred-1
+  cloud: test-cloud
+controllers:
+- name: controller-1
+  uuid: 00000001-0000-0000-0000-000000000001
+  cloud: test-cloud
+  region: test-cloud-region
+models:
+- name: model-1
+  uuid: 00000002-0000-0000-0000-000000000001
+  controller: controller-1
+  cloud: test-cloud
+  region: test-cloud-region
+  cloud-credential: cred-1
+  owner: alice@canonical.com
+users:
+- username: alice@canonical.com
+  access: admin
+`
+
+func (s *httpProxySuite) SetUpTest(c *gc.C) {
+	s.JIMMSuite.SetUpTest(c)
+	ctx := context.Background()
+	tester := jimmtest.GocheckTester{C: c}
+	env := jimmtest.ParseEnvironment(tester, testEnv)
+	env.PopulateDB(tester, s.JIMM.Database)
+	user, err := s.JIMM.FetchIdentity(ctx, env.Users[0].Username)
+	c.Assert(err, gc.IsNil)
+	err = user.SetModelAccess(ctx, names.NewModelTag(env.Models[0].UUID), ofganames.AdministratorRelation)
+	c.Assert(err, gc.IsNil)
+	model := &dbmodel.Model{UUID: sql.NullString{String: env.Models[0].UUID, Valid: true}}
+	err = s.JIMM.Database.GetModel(ctx, model)
+	c.Assert(err, gc.IsNil)
+	s.model = model
+	err = s.JIMM.GetCredentialStore().PutControllerCredentials(ctx, model.Controller.Name, "user", "psw")
+	c.Assert(err, gc.IsNil)
+}
+
+func (s *httpProxySuite) TestCharmHTTPServe(c *gc.C) {
+	ctx := context.Background()
+	httpProxier := jujuapi.HTTPProxier{JIMM: s.JIMM}
+	expectU, expectP, err := s.JIMM.GetCredentialStore().GetControllerCredentials(ctx, s.model.Controller.Name)
+	c.Assert(err, gc.IsNil)
+	// we expect the controller to respond with TLS
+	fakeController := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		u, p, _ := r.BasicAuth()
+		c.Assert(u, gc.Equals, names.NewUserTag(expectU).String())
+		c.Assert(p, gc.Equals, expectP)
+		_, err = w.Write([]byte("OK"))
+		c.Assert(err, gc.IsNil)
+	}))
+	defer fakeController.Close()
+	controller := s.model.Controller
+	pemData := pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: fakeController.Certificate().Raw,
+	})
+	controller.CACertificate = string(pemData)
+
+	tests := []struct {
+		description    string
+		setup          func()
+		url            string
+		modelUUID      string
+		statusExpected int
```

Comment:
Can you please add a `expectedBody` field to assert it against the returned HTTP response body?